src/common/playlists/PlayList.jsx

```diff
@@ -23,8 +32,12 @@ const PlayList = () => {
   if (plays?.length === 0) {
     return (
       <div className='play-not-found'>
-        <ImageOops className='play-not-found-image' />
-        <p className='page-404-lead'>Play not found</p>
+        <ImageOops className='play-not-found-image' />  
+        <p className='page-404-lead'>Play not found 
+        {hasFilterQuery? " for matched filter criteria" : null}
+        {hasFilterQuery && searchTerm ? " and": null}
+        {searchTerm ? ` for  ${searchTerm}` : null}
```

Comment:
A tiny observation,
This is a little complex from the readability front. Can you simplify it? Something like creating a method for doing this job in a more readable way?